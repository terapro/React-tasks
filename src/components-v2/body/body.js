import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import {Header} from "src/components-v2/header/header";
import {Footer} from "src/components-v2/footer/footer";
import {BodySection} from "src/components-v2/body/section/body-section";
import { withRouter } from 'react-router'

export const Body = withRouter(({children}) => (
  <Fragment>
      <div className="content">
        {children}
      </div>
      <BodySection bgDark asFooter>
        <Footer/>
      </BodySection>
  </Fragment>
));

Body.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

Body.defaultProps = {
  children: null
};
