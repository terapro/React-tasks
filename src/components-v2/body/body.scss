* {
  margin: 0;
  padding: 0;
}

html,
body {
  height: 100%;
  font-family: $fontFamily;
  font-size: $fontSizeDefault;
  background: $color_black;
}


div.doc-wrapper {
  display: flex;
  flex-direction: column;
  height: 100%;
}

div.content {
  flex: 1 0 auto;
}