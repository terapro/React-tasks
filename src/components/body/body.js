import React, {Fragment} from 'react';

import PropTypes from 'prop-types';

import {Header} from 'src/components/header/header.js';
import {Results} from 'src/components/results/results.js';
import {Footer} from 'src/components/footer/footer.js';

const Body = (props) => {
  const {startSearch, searchMode, filmMode, filmInfo, setSearchModeCallback, searchResult, setFilmMode,
    filmModeGenre, searchPhrase, searchInFilmModeByGenre} = props;
  return (
    <Fragment>
      <div className="wrapper">
        <Header startSearch={startSearch}
                searchMode={searchMode}
                filmMode={filmMode}
                filmInfo={filmInfo}
                setSearchModeCallback={setSearchModeCallback}
        />
        <Results searchInFilmModeByGenre={searchInFilmModeByGenre}
                 searchResult={searchResult}
                 setFilmMode={setFilmMode}
                 filmModeGenre={filmModeGenre}
                 filmMode={filmMode}
                 searchPhrase={searchPhrase}
        />
      </div>
      <Footer/>
    </Fragment>
  )
};

Body.propTypes = {
  searchPhrase: PropTypes.string.isRequired,
  filmModeGenre: PropTypes.string,
  startSearch: PropTypes.func.isRequired,
  searchResult: PropTypes.array.isRequired,
  setFilmMode: PropTypes.func.isRequired,
  searchMode: PropTypes.bool,
  filmMode: PropTypes.bool,
  filmInfo:  PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    tagline: PropTypes.string,
    vote_average: PropTypes.number,
    vote_count: PropTypes.number,
    release_date: PropTypes.string,
    poster_path: PropTypes.string,
    overview: PropTypes.string,
    budget: PropTypes.number,
    revenue: PropTypes.number,
    genres: PropTypes.arrayOf(PropTypes.string),
    runtime: PropTypes.number
  }),
  setSearchModeCallback: PropTypes.func.isRequired,
  searchInFilmModeByGenre: PropTypes.func.isRequired
};
Body.defaultProps = {
  searchMode: true,
  filmMode: false,
  filmModeGenre: ''
};


export {Body};