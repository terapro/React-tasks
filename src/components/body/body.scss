* {
  margin: 0;
  padding: 0;
}

html,
body {
  height: 100%;
  font-family: $fontFamily;
  font-size: $fontSizeDefault;
}

.doc-wrapper {
  display: flex;
  flex-direction: column;
  height: 100%;
}

div.wrapper {
  width: $appWidth;
  margin: 0 auto;
  flex: 1 0 auto;
}