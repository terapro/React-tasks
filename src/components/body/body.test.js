import React from 'react';
import {Body} from 'src/components/body/body.js';
import {shallow} from 'enzyme';

const mockFilm = {
  "id": 333339,
  "title": "Ready Player One",
  "tagline": "A better reality awaits.",
  "vote_average": 8.1,
  "vote_count": 617,
  "release_date": "2018-03-28",
  "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
  "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
  "budget": 175000000,
  "revenue": 0,
  "genres": [
    "Adventure",
    "Science Fiction",
    "Action"
  ],
  "runtime": 140
};


const mockProps = {
  searchPhrase: 'Searching something',
  filmModeGenre: 'action',
  startSearch: jest.fn(),
  searchResult: [mockFilm],
  setFilmMode: jest.fn(),
  searchMode: true,
  filmMode: false,
  filmInfo: mockFilm,
  setSearchModeCallback: jest.fn(),
  searchInFilmModeByGenre: jest.fn()
};

const mockPropsForFilmMode = {...mockProps, ...{filmMode: true,  searchMode: false}};
const mockPropsNoSearchResult = {...mockProps, ...{searchResult: []}};

describe('<Body />', () => {
  describe('Rendering', () => {
    it('should render the Body in the search mode', () => {
      const wrapper = shallow(<Body {...mockProps} />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render the Body in the film mode', () => {
      const wrapper = shallow(<Body {...mockPropsForFilmMode} />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render the Body with empty search result in search mode', () => {
      const wrapper = shallow(<Body {...mockPropsNoSearchResult} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
