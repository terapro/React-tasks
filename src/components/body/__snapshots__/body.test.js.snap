// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Body /> Rendering should render the Body in the film mode 1`] = `
<React.Fragment>
  <div
    className="wrapper"
  >
    <Header
      filmInfo={
        Object {
          "budget": 175000000,
          "genres": Array [
            "Adventure",
            "Science Fiction",
            "Action",
          ],
          "id": 333339,
          "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
          "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
          "release_date": "2018-03-28",
          "revenue": 0,
          "runtime": 140,
          "tagline": "A better reality awaits.",
          "title": "Ready Player One",
          "vote_average": 8.1,
          "vote_count": 617,
        }
      }
      filmMode={true}
      searchMode={false}
      setSearchModeCallback={[MockFunction]}
      startSearch={[MockFunction]}
    />
    <Results
      filmMode={true}
      filmModeGenre="action"
      searchInFilmModeByGenre={[MockFunction]}
      searchPhrase="Searching something"
      searchResult={
        Array [
          Object {
            "budget": 175000000,
            "genres": Array [
              "Adventure",
              "Science Fiction",
              "Action",
            ],
            "id": 333339,
            "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
            "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
            "release_date": "2018-03-28",
            "revenue": 0,
            "runtime": 140,
            "tagline": "A better reality awaits.",
            "title": "Ready Player One",
            "vote_average": 8.1,
            "vote_count": 617,
          },
        ]
      }
      setFilmMode={[MockFunction]}
      startSearch={[Function]}
    />
  </div>
  <Footer />
</React.Fragment>
`;

exports[`<Body /> Rendering should render the Body in the search mode 1`] = `
<React.Fragment>
  <div
    className="wrapper"
  >
    <Header
      filmInfo={
        Object {
          "budget": 175000000,
          "genres": Array [
            "Adventure",
            "Science Fiction",
            "Action",
          ],
          "id": 333339,
          "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
          "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
          "release_date": "2018-03-28",
          "revenue": 0,
          "runtime": 140,
          "tagline": "A better reality awaits.",
          "title": "Ready Player One",
          "vote_average": 8.1,
          "vote_count": 617,
        }
      }
      filmMode={false}
      searchMode={true}
      setSearchModeCallback={[MockFunction]}
      startSearch={[MockFunction]}
    />
    <Results
      filmMode={false}
      filmModeGenre="action"
      searchInFilmModeByGenre={[MockFunction]}
      searchPhrase="Searching something"
      searchResult={
        Array [
          Object {
            "budget": 175000000,
            "genres": Array [
              "Adventure",
              "Science Fiction",
              "Action",
            ],
            "id": 333339,
            "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
            "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
            "release_date": "2018-03-28",
            "revenue": 0,
            "runtime": 140,
            "tagline": "A better reality awaits.",
            "title": "Ready Player One",
            "vote_average": 8.1,
            "vote_count": 617,
          },
        ]
      }
      setFilmMode={[MockFunction]}
      startSearch={[Function]}
    />
  </div>
  <Footer />
</React.Fragment>
`;

exports[`<Body /> Rendering should render the Body with empty search result in search mode 1`] = `
<React.Fragment>
  <div
    className="wrapper"
  >
    <Header
      filmInfo={
        Object {
          "budget": 175000000,
          "genres": Array [
            "Adventure",
            "Science Fiction",
            "Action",
          ],
          "id": 333339,
          "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
          "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
          "release_date": "2018-03-28",
          "revenue": 0,
          "runtime": 140,
          "tagline": "A better reality awaits.",
          "title": "Ready Player One",
          "vote_average": 8.1,
          "vote_count": 617,
        }
      }
      filmMode={false}
      searchMode={true}
      setSearchModeCallback={[MockFunction]}
      startSearch={[MockFunction]}
    />
    <Results
      filmMode={false}
      filmModeGenre="action"
      searchInFilmModeByGenre={[MockFunction]}
      searchPhrase="Searching something"
      searchResult={Array []}
      setFilmMode={[MockFunction]}
      startSearch={[Function]}
    />
  </div>
  <Footer />
</React.Fragment>
`;
